### PR TITLE
Refactor

### LINKED ISSUE #131: Refactor
The app has been growing organically for a while and could use a decent refactor. Its hard to point to any specific place within the app where this should be done, but overall the code quality code be improved.

### PR DESCRIPTION
**Pull request by dfulu**
# Pull Request

## Description

A medium refactor of the code which keeps all of same the functionality and user facing APIs of the old code

Extra fixes:
1. We use elevation from the batch to compute night time masks. The elevation values from `ocf-data-sampler` are 0-1 scaled whilst the elevation values from `ocf_datapipes` are standardised. A fix for this, including legacy flag was added
2. In the pvnet pipeline in ocf_datapipes with the production flag switched on, when satellite data is missing we allow linear interpolation to fill gaps of a few steps. This is not included in `ocf-data-sampler`. We move this satellite interpolation into the app. This will work with both the legacy and new dataloader, and having this step in the app gives us the opportunity to take more care in the interpolation and monitor this step more closely

Closes #131 

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvnet_app/app.py [hunk 1]
```diff
@@ -1,13 +1,4 @@
...
-
-This app expects these evironmental variables to be available:
-    - DB_URL
-    - NWP_UKV_ZARR_PATH
-    - NWP_ECMWF_ZARR_PATH
-    - SATELLITE_ZARR_PATH
-    - RUN_EXTRA_MODELS
-    - USE_ADJUSTER
-    - SAVE_GSP_SUM
+"""App to run inference for PVNet models
```
Comment (peterdudfield) [inline pvnet_app/app.py:1]: Could you add the relevant env vars bk in please

File: pvnet_app/app.py [hunk 2]
```diff
@@ -305,18 +292,14 @@ def app(
...
-    if use_satellite:
-        logger.info("Downloading satellite data")
-        download_all_sat_data()
-
-        # Preprocess the satellite data and record the delay of the most recent non-nan timestep
-        all_satellite_datetimes, data_freq_minutes = preprocess_sat_data(
-            t0, 
-            use_legacy=use_day_ahead_model
-        )
+    logger.info("Downloading satellite data")
```
Comment (peterdudfield) [inline pvnet_app/app.py:295]: I added in this use satellite, as it didn't work when there was no satellite data. 

I wonder if there is an easy way to add this functionality back in

Comment (peterdudfield) [reply pvnet_app/app.py:295]: Actually maybe you keep this in

File: pvnet_app/data/satellite.py [hunk 3]
```diff
@@ -89,88 +80,179 @@ def combine_5_and_15_sat_data(t0) -> tuple[int, pd.DatetimeIndex]:
...
 
+def fill_1d_bool_gaps(x, max_gap):
+    """In a boolean array, fill consecutive False elements if their number is less than the gap_size
+
+    Args:
+        x: A 1-dimensional boolean array
+        max_gap: integer of the maximum gap size which will be filled with True
+
+    Returns:
+        A 1-dimensional boolean array
```
Comment (peterdudfield) [inline pvnet_app/data/satellite.py:124]: Good description

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
Perhaps double checking it still works if there is no satellite data at all. 

This is our back up where we just run tge ecmwf model only

File: pvnet_app/app.py (hunk 1 above)
Comment (dfulu) [reply pvnet_app/app.py:1]: The environmental vars are also in the docstring for the `app()` function. I thought it was messy to have them in two places in the one file. Plus these are now out of date. What do you think?

Comment (peterdudfield) [reply pvnet_app/app.py:1]: if they are up to date in the doce string, then maybe thats ok. Are they up to date on the readme too?
Its probably worth adding them in on the readme, but that could be a seperate issue if you like

Comment (dfulu) [reply pvnet_app/app.py:1]: Yeh they are up to date in the docstring. We've never had them (or any usage info) in the README, so I think if we want that it should be a separate issue

**Comment by dfulu** [comment]
> Perhaps double checking it still works if there is no satellite data at all.
> 
> This is our back up where we just run tge ecmwf model only

I've added a test for the case where there is no satellite data avilable

**Pull request merged**